### ISSUE TITLE
Unsatisfying results for image detection problem

### ISSUE THREAD

**Issue by delhomer** [Oslandia/mapillary_classification#2]
After a few training iterations, the model converge towards a situation such that popular labels (largely represented in the dataset) are always set as 'true', whilst rare labels are always set as 'false'.

**Comment by delhomer** [comment]
Some tracks:
- weight the label contributions to loss
- tune the learning rate
- change the optimizer (Adam)
- improve image preprocessing (+ data augmentation? data reduction?)
- build a more complex network

**Comment by delhomer** [comment]
After some tests with a simpler network, it appears that model training must be undertaken for a far larger number of epochs (20 instead of 5, actually, to get satisfying results on a very simple network).

Tests have to be continued on Mapillary dataset to provide a more precise picture of this solution.

**Comment by delhomer** [comment]
New instance processing show fairly good training performances for Shapes dataset (>99% validation accuracy with a `vgg16` network). As soon as new runs on Mapillary dataset confirm this point, this issue can be closed.

**Comment by delhomer** [comment]
The issue seems addressed, at least regarding the very bad results a few month ago. Now neural networks can beat the label expected probabilities deduced from their popularity. We can close the issue, and reopen a new better-defined one later if needed.
